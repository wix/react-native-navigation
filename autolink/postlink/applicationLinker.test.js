import fs from 'fs';

/**
 * Mocks
 */

jest.mock('../postlink/log', () => ({
    log: console.log,
    logn: console.log,
    warn: console.log,
    warnn: console.log,
    info: console.log,
    infon: console.log,
    debug: console.log,
    debugn: console.log,
    errorn: console.log,
}));

/**
 * Tests
 */

describe('applicationLinker', () => {
    beforeEach(() => { });

    it('should work for RN 0.77', () => {
        jest.mock('../postlink/path', () => {
            const { copyFileSync } = require('fs');
            const { tmpdir } = require('os');
            const path = require('path');

            const tmpMainApplicationPath = path.resolve(tmpdir(), 'rnn-tests_MainApplication.kt');

            copyFileSync(
                path.resolve('autolink/fixtures/rn77/MainApplication.kt.template'),
                tmpMainApplicationPath
            );

            return {
                mainApplicationKotlin: tmpMainApplicationPath,
            };
        });

        const ApplicationLinker = require('./applicationLinker');
        const linker = new ApplicationLinker();

        linker.link();
        const mainApplicationContent = fs.readFileSync(linker.applicationPath, 'utf8');
        expect(mainApplicationContent).toMatchSnapshot();
    });
}); 