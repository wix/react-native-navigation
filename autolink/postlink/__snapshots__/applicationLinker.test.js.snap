// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applicationLinker should work for RN 0.77 1`] = `
"package com.app

import android.app.Application
import com.facebook.react.PackageList
import com.reactnativenavigation.NavigationApplication
import com.facebook.react.ReactHost
import com.facebook.react.ReactNativeHost
import com.reactnativenavigation.react.NavigationReactNativeHost
import com.facebook.react.ReactPackage
import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.load
import com.facebook.react.defaults.DefaultReactHost.getDefaultReactHost
import com.facebook.react.defaults.DefaultReactNativeHost
import com.facebook.react.soloader.OpenSourceMergedSoMapping
import com.facebook.soloader.SoLoader

class MainApplication : NavigationApplication() {

  override val reactNativeHost: ReactNativeHost =
      object : DefaultNavigationReactNativeHost(this) {
        override fun getPackages(): List<ReactPackage> =
            PackageList(this).packages.apply {
              // Packages that cannot be autolinked yet can be added manually here, for example:
              // add(MyReactNativePackage())
            }

        override fun getJSMainModuleName(): String = "index"

        override fun getUseDeveloperSupport(): Boolean = BuildConfig.DEBUG

        override val isNewArchEnabled: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
        override val isHermesEnabled: Boolean = BuildConfig.IS_HERMES_ENABLED
      }

  override val reactHost: ReactHost
    get() = getDefaultReactHost(applicationContext, reactNativeHost)

  override fun onCreate() {
    super.onCreate()
    
    
  }
}
"
`;
